pkg/commatrix/commatrix_test.go

```diff
@@ -0,0 +1,217 @@
+)
+
+var (
+	infra = &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			ControlPlaneTopology: configv1.HighlyAvailableTopologyMode,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+			},
+		},
+	}
+
+	testNode = &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-node",
+			Labels: map[string]string{
+				"node-role.kubernetes.io/master": "",
+			},
+		},
+	}
+
+	testPod = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-app-pod",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				"kubernetes.io/service-name": "test-service",
+				"app":                        "test-app",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "test-container",
+					Image: "test-image:latest",
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 80,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testService = &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-service",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				"kubernetes.io/service-name": "test-service",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "test-app",
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port:     80,
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Type: corev1.ServiceTypeNodePort,
+		},
+	}
+
+	testEndpointSlice = &discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-service-endpoints",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				"kubernetes.io/service-name": "test-service",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: "Service",
+					Name: "test-service",
+				},
+			},
+		},
+		AddressType: discoveryv1.AddressTypeIPv4,
+		Endpoints: []discoveryv1.Endpoint{
+			{
+				NodeName:  &testNode.Name,
+				Addresses: []string{"192.168.1.1", "192.168.1.2"},
+			},
+		},
+		Ports: []discoveryv1.EndpointPort{
+			{
+				Name:     nil,
+				Port:     func(i int32) *int32 { return &i }(80),
+				Protocol: func(p corev1.Protocol) *corev1.Protocol { return &p }(corev1.ProtocolTCP),
+			},
+		},
+	}
+
+	testEpsComDetails = []types.ComDetails{
+		{
+			Direction: "Ingress",
+			Protocol:  "TCP",
+			Port:      80,
+			Namespace: "test-ns",
+			Service:   "test-service",
+			Pod:       "test-app-pod",
+			Container: "test-container",
+			NodeRole:  "master",
+			Optional:  false,
+		},
+	}
+)
+
+func TestCommatrixGeneration(t *testing.T) {
+	expectedComDetails := slices.Concat(testEpsComDetails, types.CloudStaticEntriesWorker, types.CloudStaticEntriesMaster,
+		types.GeneralStaticEntriesMaster, types.GeneralStaticEntriesWorker, types.StandardStaticEntries)
+
+	expectedComMatrix = types.ComMatrix{Matrix: expectedComDetails}
+	expectedComMatrix.SortAndRemoveDuplicates()
+
+	testCases := []struct {
```

Comment:
Should also check failures and/or validation func

Reply:
i added more